controllers/admin/AdminImagesController.php

```diff
@@ -32,18 +32,171 @@ class AdminImagesControllerCore extends AdminController
     protected $start_time = 0;
     protected $max_execution_time = 7200;
     protected $display_move;
+    protected $generateAdditionalAvif;
 
     public function __construct()
     {
         $this->bootstrap = true;
         $this->table = 'image_type';
         $this->className = 'ImageType';
-        $this->lang = false;
 
+        parent::__construct();
+    }
+
+    public function init()
+    {
         $this->addRowAction('edit');
         $this->addRowAction('delete');
 
-        parent::__construct();
+        parent::init();
+
+        $this->generateAdditionalAvif = $this->get('prestashop.core.configuration.avif_extension_checker')->isAvailable();
+
+        $imageFormatsDisabled = [];
+
+        if (false === $this->generateAdditionalAvif) {
+            $imageFormatsDisabled['avif'] = true;
+        }
+
+        $fields = [
+            'PS_IMAGE_FORMAT' => [
```

Comment:
This should be reverted to this as per discussion.
![img_4390](https://user-images.githubusercontent.com/6097524/208086948-2c6a2dbb-171c-4fdc-894e-3c2b64418b28.jpg)

AVIF and WEBP checkboxes should control the booleans `PS_ADDITIONAL_IMAGE_WEBP` and `PS_ADDITIONAL_IMAGE_AVIF`.